### PR TITLE
CPLAT-4160 v2.0.0 (Dart 2)

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Description
Merging the integration branch into master. The Dart1 master branch is now `master_dart1`.

All of the changes here have been previously reviewed and tested in one of these PRs:
https://github.com/Workiva/over_react/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+base%3A2.0.0-dev

## Testing suggestions:
- [ ] CI passes
- [ ] Some quick smoke tests!

## Potential areas of regression:
Yeah for sure for sure.

---

> __FYA:__ @greglittlefield-wf @aaronlademann-wf @kealjones-wk @maxwellpeterson-wf @corwinsheahan-wf @sebastianmalysa-wf @smaifullerton-wk @georgelesica-wf @robbecker-wf

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/over_react/component_declaration/builder_integration_tests/accessor_mixin_integration_test.dart [hunk 1]
```diff
@@ -147,7 +146,10 @@ main() {
 
 
 @PropsMixin()
-abstract class TestPropsMixin {
+abstract class _$TestPropsMixin {
+  // To ensure the codemod regression checking works properly, please keep this
```
Comment (corwinsheahan-wf) [inline test/over_react/component_declaration/builder_integration_tests/accessor_mixin_integration_test.dart]: need to remove this. I'm pushing a commit.

File: .gitignore [hunk 2]
```diff
@@ -35,6 +35,7 @@ _site
 packages
 .packages
 .dart_tool
+*.g.dart
```
Comment (greglittlefield-wf) [inline .gitignore]: Can we remove this? I would think that any `.g.dart` files that are created outside of `.dart_tool` should be tracked.

Comment (corwinsheahan-wf) [reply .gitignore]: +1

**Comment by corwinsheahan-wf** [comment]
Note to reviewers: Added a second builder which uses the same factory as the `overReactBuilder`, but it builds to source and is an "opt-in" builder, meaning it won't affect dependents unless they explicitly utilize it. We configure this builder to apply to the over_react package, and disable the `overReactBuilder` which builds to cache. What this means is that, for `over_react`, the `.over_react.g.dart` files will be build to source and committed, but dependents of over_react will still build to cache. 

Here's the commit to update the build config: [58da397](https://github.com/Workiva/over_react/pull/240/commits/58da397a5a827cf304d54dcbd8a089123b7f4bed)

File: build.yaml [hunk 3]
```diff
@@ -6,14 +6,25 @@ builders:
...
     builders:
-      over_react|overReactBuilder:
+      over_react|overReactLocalBuilder:
+        enabled: true
         generate_for:
           exclude:
             # This tests un-built behavior, and therefore should not be built
             - "test/over_react/component_declaration/builder_helpers_test.dart"
+      over_react|overReactBuilder:
+        enabled: false
```
Comment (evanweible-wf) [inline build.yaml]: So this only disables it for this package? Is that how `targets.$default` works?

Comment (corwinsheahan-wf) [reply build.yaml]: Yes, that only does this for this package. The `$default` target specifies the behavior that consumers will use when running builds on this package: https://github.com/dart-lang/build/tree/master/build_config#dividing-a-package-into-build-targets

Comment (corwinsheahan-wf) [reply build.yaml]: Consuming packages' builds are unaffected by that (only applies to over_react)

**Comment by aaronlademann-wf** [comment]
@corwinsheahan-wf @evanweible-wf does this need the RM-friendly ticket number / "Release over_react 2.0.0" added to the title in order for MARV to pick it up?

**Comment by evanweible-wf** [comment]
+1

File: pubspec.yaml [hunk 4]
```diff
@@ -6,51 +6,36 @@ authors:
...
-  dart_dev: ^1.9.6
+  dart_dev: ^2.0.0
   dependency_validator: ^1.2.2
-  mockito: ^2.2.2
-  over_react_test: ^1.5.0
-  source_gen: ^0.7.4+3
-  test: ^0.12.34
-
-transformers:
-  - over_react:
```
Comment (greglittlefield-wf) [inline pubspec.yaml]: Goodbye old friend 👋

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passes
- [x] Smoke tests look good

@Workiva/release-management-p

**Pull request merged**